cms/static/js/views/uploads.js

```diff
@@ -26,11 +26,43 @@ define(["jquery", "underscore", "gettext", "js/views/modals/base_modal", "jquery
                 BaseModal.prototype.addActionButtons.call(this);
             },
 
+            loadImageFile: function (file, callback) {
+                // Load an image file.
+                var dimensions = this.model.get('dimensions'),
+                    image = new Image();
+                image.onload = function() { callback(image) };
+                image.src = URL.createObjectURL(file);
+            },
+
             renderContents: function() {
-                var isValid = this.model.isValid(),
-                    selectedFile = this.model.get('selectedFile'),
-                    oldInput = this.$("input[type=file]").get(0);
-                BaseModal.prototype.renderContents.call(this);
+                var selectedFile = this.model.get('selectedFile');
+                var mimeTypes = this.model.get('mimeTypes');
+                var self = this;
+
+                function isSelectedFileAnImage() {
```

Comment:
`isSelectedFileAnImage` should check only the mime type of `selected_file` as its name suggests.